tests/Nidup/Architool/Infrastructure/Filesystem/FolderReferenceUpdaterTest.php

```diff
@@ -0,0 +1,136 @@
+                                ],
+                                'Pim' => [
+                                    'Platform' => [
+                                        'Domain' => [
+                                            'Batch' => []
+                                        ]
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ];
+
+        $this->root = vfsStream::setup('root', null, $rootStructure);
+    }
+
+    public function testNamespaceDeclarationIsUpdatedInPhp()
+    {
+        $filePath =  $this->root->url().'/My/Project/Dir/src/Akeneo/Component/Batch/MyClassInBatchComponent.php';
+        file_put_contents(
+            $filePath,
+            'namespace Akeneo\Component\Batch;'
+        );
+
+        $folder = new Folder(new FolderNamespace('Akeneo/Component/Batch'));
+        $folder->move(new FolderNamespace('Akeneo/Pim/Platform/Domain/Batch'));
+
+        $updater = new FolderReferenceUpdater($this->root->url().'/My/Project/Dir');
+
+        $updater->update($folder);
+
+        $this->assertEquals(
+            'namespace Akeneo\Pim\Platform\Domain\Batch;',
+            file_get_contents($filePath)
+        );
+    }
+
+    public function testDifferentNamespaceDeclarationWithSamePrefixIsNotUpdatedInPhp()
+    {
+        $filePath =  $this->root->url().'/My/Project/Dir/src/Akeneo/Component/BatchQueue/MyClassInBatchQueueComponent.php';
+        $namespaceDeclaration = 'namespace Akeneo\Component\BatchQueue;';
+        file_put_contents(
+            $filePath,
+            $namespaceDeclaration
+        );
+
+        $folder = new Folder(new FolderNamespace('Akeneo/Component/Batch'));
+        $folder->move(new FolderNamespace('Akeneo/Pim/Platform/Domain/Batch'));
+
+        $updater = new FolderReferenceUpdater($this->root->url().'/My/Project/Dir');
+
+        $updater->update($folder);
+
+        $this->assertEquals(
+            $namespaceDeclaration,
+            file_get_contents($filePath)
+        );
+    }
+
+    public function testNamespaceReferenceIsUpdatedInPhp()
+    {
+        $filePath =  $this->root->url().'/My/Project/Dir/src/MyFileUsingBatch.php';
+        file_put_contents(
+            $filePath,
+            'use Akeneo\Component\Batch\SomeClassFromBatchComponent;'
+        );
+
+        $folder = new Folder(new FolderNamespace('Akeneo/Component/Batch'));
+        $folder->move(new FolderNamespace('Akeneo/Pim/Platform/Domain/Batch'));
+
+        $updater = new FolderReferenceUpdater($this->root->url().'/My/Project/Dir');
+
+        $updater->update($folder);
+
+        $this->assertEquals(
+            'use Akeneo\Pim\Platform\Domain\Batch\SomeClassFromBatchComponent;',
+            file_get_contents($filePath)
+        );
+    }
+
+    public function testDifferentNamespaceReferenceWithSamePrefixIsNotUpdatedInPhp()
```

Comment:
idem

Reply:
I test if a different namespace reference with same prefix is not updated.